### ISSUE TITLE
AD: joining to domain doesn't work

### ISSUE THREAD

**Issue by brusch** [rockstor/rockstor-core#860]
I've filled our all the required fields for the AD integration. 
But when I click on "Join Domain" nothing is shown, ... no errors in the logs or in the JS console. 

When I try to enable the service first i get an error message. In the logs I've the following contents: 

```
Sep 04 09:58:21 rockstor systemd[1]: winbind.service: Supervising process 31815 which is not our child. We'll most likely not notice when it exits.
Sep 04 09:58:21 rockstor winbindd[31815]: [2015/09/04 09:58:21.806824,  0] ../source3/winbindd/winbindd_cache.c:3230(initialize_winbindd_cache)
Sep 04 09:58:21 rockstor winbindd[31815]: initialize_winbindd_cache: clearing cache and re-creating with version number 2
Sep 04 09:58:21 rockstor winbindd[31815]: [2015/09/04 09:58:21.807822,  0] ../source3/winbindd/winbindd_util.c:634(init_domain_list)
Sep 04 09:58:21 rockstor winbindd[31815]: Could not fetch our SID - did we join?
Sep 04 09:58:21 rockstor systemd[1]: winbind.service: main process exited, code=exited, status=1/FAILURE
Sep 04 09:58:21 rockstor systemd[1]: Failed to start Samba Winbind Daemon.
Sep 04 09:58:21 rockstor systemd[1]: Unit winbind.service entered failed state.
```

so it seems I've to joint to the domain first, but this doesn#T work ;-) 

see also: 

![image](https://cloud.githubusercontent.com/assets/142037/9679259/57011584-52ed-11e5-85c3-e8d1a4523421.png)

![image](https://cloud.githubusercontent.com/assets/142037/9679266/63e6fb56-52ed-11e5-8589-6eacb168fea1.png)

![image](https://cloud.githubusercontent.com/assets/142037/9679271/6db09070-52ed-11e5-9934-950ff23ba51c.png)

![image](https://cloud.githubusercontent.com/assets/142037/9679275/7bc601ea-52ed-11e5-99fb-855df524409d.png)

If you need any further information - just let me know.

**Comment by phillxnet** [comment]
@schakrava is this issue related to the ongoing forum thread:- http://forum.rockstor.com/t/binding-to-active-directoy/100/10 and your recent wikified post on the same http://forum.rockstor.com/t/how-rockstor-works-with-active-directory/413
@brusch Thanks for reporting this, it looks to me like the above ongoing issue linked above. Please have a look as I think there is current activity on that thread and wiki post.

**Comment by schakrava** [comment]
Fixed by #952

**Comment by brusch** [comment]
Thanks!
